lib/puppet/face/help.rb

```diff
@@ -146,7 +146,9 @@ def all_application_summaries()
       if (is_face_app?(appname))
         begin
           face = Puppet::Face[appname, :current]
-          result << [appname, face.summary]
+          # Add deprecation message to summary if the face is deprecated
+          summary = face.respond_to?(:deprecated?) && face.deprecated? ? face.summary + _(" (Deprecated)") : face.summary
```

Comment:
should be fine to remove `respond_to?` - this file requires `puppet/face` which requires and then defines itself as `puppet/interface`, which publicly defines `deprecated?`.